### PR TITLE
feat: prevent accidental calls in large groups (WPB-4293)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-4293" title="WPB-4293" target="_blank">StoryWPB-4293</a>  [Android] Prevent accidental calls in large groups
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Description

For large groups (>5 participants), the app will display a confirmation dialog before starting a call to prevent accidental clicks

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

Start a call in a large group ( more than 4 participants)

### Attachments (Optional)

Wire 2023-09-12 at 11_36 AM

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [697](https://10.10.124.134/job/AR-build-pipeline/697/) **failed**.

**Comment by AndroidBob** [comment]
Build [699](https://10.10.124.134/job/AR-build-pipeline/699/) **failed**.

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -761,6 +761,9 @@
     <string name="calling_ongoing_double_tap_to_go_back">Double tap to go back</string>
     <string name="calling_feature_unavailable_title_alert">Feature unavailable</string>
     <string name="calling_feature_unavailable_message_alert">The option to initiate a conference call is only available in the paid version of Wire.</string>
+    <string name="calling_confirm__start_call_title_dialog">Start a call</string>
```
Comment (alexandreferris) [inline app/src/main/res/values/strings.xml]:
```suggestion
    <string name="calling_confirm_start_call_title_dialog">Start a call</string>
```

Comment (alexandreferris) [reply app/src/main/res/values/strings.xml]: **nitpick**: just removing the extra `_`

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [708](https://10.10.124.134/job/AR-build-pipeline/708/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/common/WireDialog.kt [hunk 2]
```diff
@@ -96,7 +97,7 @@ fun WireDialog(
         optionButton2Properties = optionButton2Properties,
         dismissButtonProperties = dismissButtonProperties,
         buttonsHorizontalAlignment = buttonsHorizontalAlignment,
-        modifier = modifier,
+        modifier = modifier.semantics(mergeDescendants = true) {},
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/common/WireDialog.kt:100]: question: what does it do exactly for the dialog? 🤔

**Review by saleniuk — APPROVED** [review]

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (ohassine) [reply app/src/main/res/values/strings.xml]: Good catch

File: app/src/main/kotlin/com/wire/android/ui/common/WireDialog.kt (hunk 2 above)
Comment (ohassine) [reply app/src/main/kotlin/com/wire/android/ui/common/WireDialog.kt:100]: This is for accessibility. When the dialog is displayed, Talkback will read all of its content not just the title

**Comment by AndroidBob** [comment]
Build [716](https://10.10.124.134/job/AR-build-pipeline/716/) **failed**.

**Comment by AndroidBob** [comment]
Build [722](https://10.10.124.134/job/AR-build-pipeline/722/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.5.0(7556307)(7556307)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.5.0(7556307)(7556307)-staging-compat.apk)

**Pull request merged**